src/preprocessor.py

```diff
@@ -787,6 +787,22 @@ def check_time_bounds(self, ds, var: translation.TranslatedVarlistEntry, freq: s
                 var.log.error(err_str)
                 raise IndexError(err_str)
 
+    def normalize_group_time_vals(self, time_vals: np.ndarray) -> np.ndarray:
+        """Apply logic to fomat time_vals lists found in 
+        check_grouo_daterange and convert them into str type.
+        This function also handles missing leading zeros
+        """
+        poss_digits = [6,8,10,12,14]
+        for i in range(len(time_vals)):
+            if isinstance(time_vals[i], str):
+                if '-' in time_vals[i]:
+                    time_vals[i] = time_vals[i].replace('-', '')
+                if ':' in time_vals[i]:
+                    time_vals[i] = time_vals[i].replace(':', '')
```

Comment:
you [can replace multiple characters in one call](https://stackoverflow.com/questions/3411771/best-way-to-replace-multiple-characters-in-a-string) without having to explicitly check for them, e.g.,
```
time_vals[i] = time_vals[i].replace('-', ':').replace('', '').
```

Reply:
I'll condense that for the final PR